airflow/www/views.py

```diff
@@ -1838,10 +1839,10 @@ def trigger(self, session=None):
                 form=form,
                 is_dag_run_conf_overrides_params=is_dag_run_conf_overrides_params,
             )
-
-        dr = DagRun.find(dag_id=dag_id, execution_date=execution_date, run_type=DagRunType.MANUAL)
+        # if run_id is not None, filter dag runs based on run id and ignore execution date
+        dr = DagRun.find_duplicate(dag_id=dag_id, run_id=run_id, execution_date=execution_date)
         if dr:
-            flash(f"This run_id {dr.run_id} already exists")
+            flash(f"The run_id {run_id} already exists", "error")
```

Comment:
```suggestion
            flash(f"The run_id {dr.run_id} already exists", "error")
```

I think this should use `dr.run_id` instead because `run_id` can be empty from the UI?